src/nimblepkg/nimscriptapi.nim

```diff
@@ -22,46 +25,157 @@ var
   foreignDeps*: seq[string] = @[] ## The foreign dependencies. Only
                                   ## exported for 'distros.nim'.
 
+  beforeHooks: seq[string] = @[]
+  afterHooks: seq[string] = @[]
+  commandLineParams: seq[string] = @[]
+  flags: TableRef[string, seq[string]]
+
+  command = "e"
+  project = ""
+  success = false
+  retVal = true
+
 proc requires*(deps: varargs[string]) =
   ## Call this to set the list of requirements of your Nimble
   ## package.
   for d in deps: requiresData.add(d)
 
+proc getParams() =
+  for i in 5 .. paramCount():
+    if paramStr(i).len != 0:
+      commandLineParams.add paramStr(i).normalize
+
+proc getCommand(): string =
+  return command
+
+proc setCommand(cmd: string, prj = "") =
+  command = cmd
+  if prj.len != 0:
+    project = prj
+
+proc switch(key: string, value="") =
+  if flags.isNil:
+    flags = newTable[string, seq[string]]()
+
+  if flags.hasKey(key):
+    flags[key].add(value)
+  else:
+    flags[key] = @[value]
+
+template `--`(key, val: untyped) =
```

Comment:
Long story but I started with `import nimscriptapi` but many of the global vars conflict with the instances in `system.nim` so nim complains to disambiguate. We cannot expect all nimble package authors to write `nimscriptapi.version`.

The only options are to `include nimscriptapi` or just have everything inline. Considering I was already checking and caching the nims files, I made nimscriptapi inline instead of having to check and write it separately. It also makes it easier to have multiple nimble processes running in parallel.

If there's a better way to get this working, I'm open to improve this.

Reply:
Does `import system except version` not work? AFAIK we don't include this file right now, so this has to work somehow

Reply:
I've implemented this but have to exclude every single nimscript var and task. Makes no sense.

```
import system except getCommand, setCommand, switch, `--`,
  packageName, version, author, description, license, srcDir, binDir, backend,
  skipDirs, skipFiles, skipExt, installDirs, installFiles, installExt, bin, foreignDeps,
  requires, task, packageName
```